### ISSUE TITLE
Make stabilize optimization more flexibl

### ISSUE THREAD

**Issue by brandonwillard** [aesara-devs/aesara#505]
We might need to expand the stabilize optimizations into distinct fixed-point steps and single-pass steps.  Forcing everything into a single fixed-point step, like we're currently doing, is not always reasonable.

_Originally posted by @brandonwillard in https://github.com/aesara-devs/aesara/issues/495#issuecomment-871666565_

**Comment by ricardoV94** [comment]
I found here they added a tag attribute to mark when a recurring optimization had already been done: https://github.com/aesara-devs/aesara/blob/391e2cfbd9ebd8403ee95591c6f654253964d0fe/aesara/tensor/math_opt.py#L2846-2850

**Comment by brandonwillard** [comment]
> I found here they added a tag attribute to mark when a recurring optimization had already been done:
> 
> https://github.com/aesara-devs/aesara/blob/391e2cfbd9ebd8403ee95591c6f654253964d0fe/aesara/tensor/math_opt.py#L2846-L2849

Yeah, I'm not sure how I feel about that approach.  It's not inherently bad, but definitely better to avoid.
